### PR TITLE
Tests for repr html

### LINKED ISSUE #1935: Tests for HTML repr
Since we removed the beautifulsoup tests, some errors have crept up in the html `repr`.
We do not want to re-introduce the bs4 tests, as they were breaking all the time and were not obvious to fix.

However, we could adopt a similar approach to the plotting tests: have some tests that take in different inputs (Variable, DataArray, Dataset, binned data, bin edges...) that test that the `repr` at least doesn't throw when being created.

In addition, we could make the factory used to create these inputs common between the html repr and the plotting tests? (and maybe also `sc.show()` and `sc.table()`?)

### PR DESCRIPTION
**Pull request by nvaytet**
- move data array and dataset factory from plotting one folder up
- add tests to check that the html repr at least does not throw in a number of cases

I find it a little annoying that we have to add some empty `__init__.py` files to trick python into thinking it is a proper module, so we can do relative imports. But I don't have a better idea.

Fixes #1935

### CODE REVIEW

File: python/tests/factory.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                          binedges=False,
+                          labels=False,
+                          masks=False,
+                          attrs=False,
+                          ragged=False,
+                          dims=None,
+                          dtype=sc.dtype.float64,
+                          unit='counts'):
+
+    dim_list = ['tof', 'x', 'y', 'z', 'Q_x']
```
Comment (SimonHeybrock) [inline python/tests/factory.py:20]: Don't use neutron-specific dim names?

File: python/tests/html_repr/html_repr_test.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+def test_html_repr_2d_binned_data_simple():
+    sc.make_html(make_binned_data_array(ndim=2))
+
+
+def test_html_repr_2d_binned_data_slice():
+    sc.make_html(make_binned_data_array(ndim=2)['x', 0])
+
+
+def test_html_repr_2d_binned_data_complex():
+    sc.make_html(make_binned_data_array(ndim=2, variances=True, masks=True))
```
Comment (SimonHeybrock) [inline python/tests/html_repr/html_repr_test.py:133]: Use pytest parametrized test for most or all in this file?

**Review by SimonHeybrock — APPROVED** [review]
Look good, thanks!

**Comment by nvaytet** [comment]
It's adding almost 400 tests. Maybe it's a little overkill?

**Comment by SimonHeybrock** [comment]
> It's adding almost 400 tests. Maybe it's a little overkill?

Is it slow?

**Comment by nvaytet** [comment]
takes just under 6 seconds, so not really

**Pull request merged**